src/3d/qgsrubberband3d.cpp

```diff
@@ -130,19 +180,46 @@ void QgsRubberBand3D::setColor( QColor color )
 {
   mColor = color;
 
-  if ( mGeometryType == Qgis::GeometryType::Line || mGeometryType == Qgis::GeometryType::Polygon )
+  if ( mMarkerType != None )
   {
-    mLineMaterial->setLineColor( color );
-    mMarkerSymbol->setColor( color.lighter( 130 ) );
+    if ( mGeometryType == Qgis::GeometryType::Line || mGeometryType == Qgis::GeometryType::Polygon )
+    {
+      mLineMaterial->setLineColor( color );
+      mMarkerSymbol->setColor( color.lighter( 130 ) );
+    }
+    else
+    {
+      mMarkerSymbol->setColor( color );
+    }
+
+    if ( mMarkerSymbol->symbolLayerCount() > 0 && mMarkerSymbol->symbolLayer( 0 )->layerType() == QLatin1String( "SimpleMarker" ) && !mOutlineColor.value() )
+    {
+      mMarkerSymbol->symbolLayer( 0 )->setStrokeColor( color );
+    }
+    updateMarkerMaterial();
   }
-  else
+
+  if ( mGeometryType == Qgis::GeometryType::Polygon )
   {
-    mMarkerSymbol->setColor( color );
+    mPolygonEntity->removeComponent( mPolygonMaterial->toMaterial( QgsMaterialSettingsRenderingTechnique::Triangles, QgsMaterialContext() ) );
```

Comment:
hmm, toMaterial() actually creates a _new_ material, so I think that removeComponent() is not going to do what you want...